docs/virtual_machines/instancetypes.md

```diff
@@ -0,0 +1,298 @@
+
+
+```yaml
+$ kubectl.sh apply -f examples/csmall.yaml -f examples/vm-cirros-csmall.yaml
+virtualmachineinstancetype.instancetype.kubevirt.io/csmall created
+virtualmachine.kubevirt.io/vm-cirros-csmall created
+
+$ kubectl get vm/vm-cirros-csmall -o json | jq .spec.instancetype
+{
+  "kind": "VirtualMachineInstancetype",
+  "name": "csmall",
+}
+
+$ virtctl start vm-cirros-csmall
+VM vm-cirros-csmall was scheduled to start
+
+$ kubectl get vm/vm-cirros-csmall -o json | jq .spec.instancetype
+{
+  "kind": "VirtualMachineInstancetype",
+  "name": "csmall",
+  "revisionName": "vm-cirros-csmall-csmall-6709b990-f717-44fe-a8ff-cb441d20b904-1"
+}
+
+$ kubectl get controllerrevision/vm-cirros-csmall-csmall-6709b990-f717-44fe-a8ff-cb441d20b904-1 -o json | jq .
+{
+  "apiVersion": "apps/v1",
+  "data": {
+    "apiVersion": "",
+    "spec": "eyJjcHUiOnsiZ3Vlc3QiOjF9LCJtZW1vcnkiOnsiZ3Vlc3QiOiIxMjhNaSJ9fQ=="
+  },
+  "kind": "ControllerRevision",
+  "metadata": {
+    "creationTimestamp": "2022-08-02T11:56:29Z",
+    "name": "vm-cirros-csmall-csmall-6709b990-f717-44fe-a8ff-cb441d20b904-1",
+    "namespace": "default",
+    "ownerReferences": [
+      {
+        "apiVersion": "kubevirt.io/v1",
+        "blockOwnerDeletion": true,
+        "controller": true,
+        "kind": "VirtualMachine",
+        "name": "vm-cirros-csmall",
+        "uid": "7ad77f47-ee51-4024-8d5b-4141aec7b04c"
+      }
+    ],
+    "resourceVersion": "7819",
+    "uid": "df3767bc-9413-4881-a961-2254f985dfd2"
+  },
+  "revision": 0
+}
+
+$ kubectl delete vm/vm-cirros-csmall
+virtualmachine.kubevirt.io "vm-cirros-csmall" deleted
+
+$ kubectl get controllerrevision/vm-cirros-csmall-csmall-6709b990-f717-44fe-a8ff-cb441d20b904-1 
+Error from server (NotFound): controllerrevisions.apps "vm-cirros-csmall-csmall-6709b990-f717-44fe-a8ff-cb441d20b904-1" not found
+
+
+```
+
+## Examples
+
+Various examples are available within the [`kubevirt`](https://github.com/kubevirt/kubevirt) repo under [`/examples`](https://github.com/kubevirt/kubevirt/tree/main/examples) including the following [CirrOS](https://github.com/cirros-dev/cirros) based example:
+
+```yaml
+---
+apiVersion: instancetype.kubevirt.io/v1alpha1
+kind: VirtualMachineInstancetype
+metadata:
+  name: clarge
+spec:
+  cpu:
+    guest: 4
+  memory:
+    guest: 2Gi
+---
+apiVersion: instancetype.kubevirt.io/v1alpha1
+kind: VirtualMachinePreference
+metadata:
+  name: virtio
+spec:
+  devices:
+    preferredDiskBus: virtio
```

Comment:
Is this doing something, since the default is `virtio` anyway?

Reply:
The default is actually `sata`.

Reply:
I was not aware, this is neither documented, nor consistent with the NIC model.
@aglitke are you aware?

I am convinced that `virtio` is a great example here.

Reply:
Reasoning is that sata works for all dominant guests (linux + win)  -speak kubevirt works out of the box with the usual guests, which is a design goal.